cpp/include/kvikio/error.hpp

```diff
@@ -75,8 +75,33 @@ struct CUfileException : public std::runtime_error {
                              cufileop_status_error(error.err)};                  \
     }                                                                            \
   } while (0)
-#define CUFILE_TRY_1(_call) CUFILE_TRY_2(_call, CUfileException)
+#define CUFILE_TRY_1(_call) CUFILE_TRY_2(_call, kvikio::CUfileException)
 #endif
 #endif
 
+#ifndef CUFILE_CHECK_STREAM_IO
+#define CUFILE_CHECK_STREAM_IO(...)                                  \
+  GET_CUFILE_CHECK_STREAM_IO_MACRO(                                  \
+    __VA_ARGS__, CUFILE_CHECK_STREAM_IO_2, CUFILE_CHECK_STREAM_IO_1) \
+  (__VA_ARGS__)
+#define GET_CUFILE_CHECK_STREAM_IO_MACRO(_1, _2, NAME, ...) NAME
+#ifdef KVIKIO_CUFILE_FOUND
+#define CUFILE_CHECK_STREAM_IO_2(_nbytes_done, _exception_type)                                    \
+  do {                                                                                             \
+    int const _nbytes = (_nbytes_done);                                                            \
+    if (_nbytes < 0) {                                                                             \
+      throw(_exception_type){std::string{"cuFile error at: "} + __FILE__ + ":" +                   \
+                             KVIKIO_STRINGIFY(__LINE__) + ": " +                                   \
+                             cufileop_status_error((CUfileOpError)(CUFILEOP_BASE_ERR - _nbytes))}; \
```

Comment:
So the documentation says that this returns:

> - The number of bytes successfully written. or;
> - -1 on IO errors. or;
> - All other errors will return a negative integer value of the `CUfileOpError` enum value. 

But this code suggests that it returns `-(some_cufile_op_error - CUFILEOP_BASE_ERR)`. Which can't be right given the above because it value-puns `CU_FILE_DRIVER_NOT_INITIALIZED` with an IO error.

Reply:
I don't think the docs are correct. I now just print the raw value of `_nbytes`. Let's fix this in a follow-up when we know the intended behavior :)